### ISSUE TITLE
Cannot select from named tuple literal

### ISSUE THREAD

**Issue by prolativ** [scala/scala3#21413]
## Compiler version

3.5.0
3.6.0-RC1-bin-20240821-a2c53a1-NIGHTLY

## Minimized code

```scala
import scala.language.experimental.namedTuples

val x = (aaa = 1).aaa
```

## Output

```scala
-- [E007] Type Mismatch Error: /tmp/NamedTuple.scala:3:18 ----------------------
  3 |val x = (aaa = 1).aaa
    |        ^^^^^^^^^^^^^
    |        Found:    Tuple1[Int]
    |        Required: (x$proxy2 : Tuple1[Int])
    |---------------------------------------------------------------------------
    |Inline stack trace
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from NamedTuple.scala:40
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from NamedTuple.scala:40
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from NamedTuple.scala:40
     ---------------------------------------------------------------------------
    |
    | longer explanation available when compiling with `-explain`
```

## Expectation

This should compile successfully

**Comment by KacperFKorban** [comment]
Looks like changing the `inline def` here into a `transparent inline def` fixes the problem, but I'm not sure if that is desired.
https://github.com/scala/scala3/blob/4293a45be76da6bb336877d56b6c7b2f972cf3e2/library/src/scala/NamedTuple.scala#L141
@aherlihy @EugeneFlesselle do you see any problems with making that change?

**Comment by soronpo** [comment]
Also, it appears there is a reporting issue it.
```
    |This location contains code that was inlined from NamedTuple.scala:40
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from NamedTuple.scala:40
    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    |This location contains code that was inlined from NamedTuple.scala:40
     ---------------------------------------------------------------------------
````
This repetition provides no additional information and seems wrong to me.

**Comment by odersky** [comment]
It looks like an Inliner bug. Basically, the inliner inlines a val x$proxy as `x$proxy2: Tuple1[Int]` and then tries
to adapt this to the expected type `x$proxy2.type`. 

Fixed by #21680
